azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -125,11 +134,13 @@
 
   private static URLClassLoader ucl;
 
-  private final RecordFactory recordFactory = RecordFactoryProvider
-      .getRecordFactory(null);
+  private final RecordFactory recordFactory = RecordFactoryProvider.getRecordFactory(null);
+  private final ExecuteAsUser executeAsUser;
 
   private HadoopSecurityManager_H_2_0(Props props)
       throws HadoopSecurityManagerException, IOException {
+    String executeAsUserBinaryFilePath = String.format("%s/%s", props.getString(NATIVE_LIB_FOLDER), "execute-as-user");
```

Comment:
Actually I updated the API to have this logic inside it. My plan is to have ExecuteAsUser as the default way of leveraging the execute-as-user binary.